### ISSUE TITLE
Uncaught ECONNREFUSED exception with Bluebird

### ISSUE THREAD

**Issue by ArenSH** [luin/ioredis#1157]
Hi, we performed some tests with cluster failover under load and got uncaught exception.

```bash
Error: connect ECONNREFUSED 172.28.1.1:6379
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '172.28.1.1',
  port: 6379
}
```

The problem seems to be that in some cases __Redis::connect__ attaches error handler to stream too late after __StandaloneConnector__ created stream.

I've reproduced this issue on Node.JS 12 with Bluebird 3.5.x. With built-in promise it seems to work, but i cannot say for sure.

In any case, current scheme with connectors and promises keeps possibility of such errors: _Stream::destroy_ already triggers error in _nextTick_ to give users opportunity to attach error listener. If because of promises error is attached later - we get uncaught exception.

One solution would be to attach "error" event handler in StandaloneConnector. Another would be to make stream instantiation synchronous.

**Comment by luin** [comment]
Hi @ArenSH , This should be fixed in v4.16.2 with https://github.com/luin/ioredis/pull/1095. What was the ioredis version?

**Comment by marcbachmann** [comment]
Let's close this. In version 5 the support for custom promise libraries also got dropped.
